### DISCUSSION TITLE
ModuleNotFoundError: No module named 'whisper'

### DISCUSSION THREAD

**Discussion by oren** [openai/whisper#1847] [General]
Note: I was not able to find a way to create an 'issue' so I hope that's the right place for my question.

I installed whisper:

```
pip3 install git+https://github.com/openai/whisper.git

Defaulting to user installation because normal site-packages is not writeable
Collecting git+https://github.com/openai/whisper.git
  Cloning https://github.com/openai/whisper.git to /tmp/pip-req-build-vs0i8qjt
  Running command git clone --filter=blob:none --quiet https://github.com/openai/whisper.git /tmp/pip-req-build-vs0i8qjt
  Resolved https://github.com/openai/whisper.git to commit e58f28804528831904c3b6f2c0e473f346223433
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting torch
Killed
```

I am trying to use it

```
python3

Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import whisper
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'whisper'
```

OS: Ubuntu 22.04.3 LTS

**Comment by phineas-pta** [comment]
failed to install whisper

**Reply by oren** [reply]
oh! thank you so much. Is there a way to troubleshoot this failure?

**Reply by phineas-pta** [reply]
re-run installation again

i saw "Killed" so likely not enough ram

**Reply by endolith** [reply]
whisper is not a dependency of llama index so it's not installed

**Comment by oren** [comment]
It looks like I have 1 GB ram free on my laptop.
```
free -g

               total        used        free      shared  buff/cache   available
Mem:               7           3           1           0           2           2
Swap:              1           1           0
```

I read on the README that 1 VRAM should be enough. I don't know if VRAM is RAM... Do I have enough memory?

**Reply by phineas-pta** [reply]
vram is gpu memory not ram, and only relevant when run whisper

free as much ram as possible to install, otherwise use something like colab

**Reply by oren** [reply]
I don't think I have GPU. Does it mean I won't be able to run whisper?

What is colab? can you share a link or description?

**Reply by ryanheise** [reply]
Whisper on CPU/RAM also works.

Since it failed specifically while installing torch, I Googled for your problem and found [this](https://stackoverflow.com/questions/62030345/pip-install-torch-killed-at-99-excessive-memory-usage). So according to that answer, you could try something like this: `pip3 install torch --no-cache-dir`. If that fails, maybe your computer doesn't have the necessary minimum resources to be able to install torch.

> What is colab?

One way to find out is to go to Google/Bing. When you type in "colab", the first page of search results tells you all about what colab is including links.

**Reply by oren** [reply]
Thanks! installing torch works.

Here is what I am trying to do, let me know if this is possible with whisper (or anything else):
I have a little CLI I wrote (in Typescript). It's a tool that manage my tasks. I want to activate some of the commands using my voice. For example: "show todos", "show dodo 12", "add todo buy milk".

Is it possible and if so, is there an example project I can look at?
If you think I should create a new 'discussion' for my question, let me know.

Thank you!

**Reply by ryanheise** [reply]
Yes, it is probably a different discussion (but you might want to search for existing discussions first).
